src/components/CustomCollection/EditableCollectionTitle.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { TextInput } from '@trussworks/react-uswds'
+
+import styles from './CustomCollection.module.scss'
+/* eslint-disable jsx-a11y/no-noninteractive-element-to-interactive-role */
+type PropTypes = {
+  text: string
+  placeholder: string
+  onSave: (s: string) => void
+}
+export const EditableCollectionTitle = ({
+  text,
+  placeholder,
+  onSave,
+}: PropTypes) => {
+  const [isEditing, setEditing] = useState(false)
+  const [currentText, setCurrentText] = useState(text)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    inputRef?.current?.focus()
+  }, [isEditing])
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    const { key } = event
+    const keys = ['Escape', 'Enter', 'Tab']
+
+    if (keys.includes(key)) {
+      setEditing(!isEditing)
+      onSave(currentText)
+    }
+  }
+
+  const handleOnBlur = () => {
+    // setEditing(!isEditing)
```

Comment:
@abbyoung I think this got commented out by accident

Reply:
grumble! fixing! thanks :)